ignite/pkg/cosmosbuf/buf.go

```diff
@@ -143,61 +156,68 @@ func (b Buf) Export(ctx context.Context, protoDir, output string) error {
 // Generate runs the buf Generate command for each file into the proto directory.
 func (b Buf) Generate(
 	ctx context.Context,
-	protoDir,
+	protoPath,
 	output,
 	template string,
-	excludeFilename ...string,
+	options ...GenOption,
 ) (err error) {
-	var (
-		excluded = make(map[string]struct{})
-		flags    = map[string]string{
-			flagTemplate:    template,
-			flagOutput:      output,
-			flagErrorFormat: fmtJSON,
-			flagLogFormat:   fmtJSON,
-		}
-	)
-	for _, file := range excludeFilename {
-		excluded[file] = struct{}{}
+	opts := genOptions{}
+	for _, apply := range options {
+		apply(&opts)
 	}
 
-	// TODO(@julienrbrt): this whole custom handling can be deleted
-	// after https://github.com/cosmos/cosmos-sdk/pull/18993 in v29.
-	if strings.Contains(protoDir, cosmosver.CosmosSDKRepoName) {
-		if b.sdkProtoDir == "" {
-			b.sdkProtoDir, err = copySDKProtoDir(protoDir)
-			if err != nil {
-				return err
-			}
-		}
-		dirs := strings.Split(protoDir, "/proto/")
-		if len(dirs) < 2 {
-			return errors.Errorf("invalid Cosmos SDK mod path: %s", dirs)
-		}
-		protoDir = filepath.Join(b.sdkProtoDir, dirs[1])
+	// find all proto files into the path.
+	foundFiles, err := xos.FindFilesExtension(protoPath, xos.ProtoFile)
+	if err != nil || len(foundFiles) == 0 {
+		return err
 	}
 
-	pkgs, err := protoanalysis.Parse(ctx, b.cache, protoDir)
-	if err != nil {
+	// check if already exist a cache for the template.
+	key, found, err := b.getCache(protoPath, template, output)
```

Comment:
I got confused here, maybe `getCache()` should have a different name, something like `copyCached()`:

 ```suggestion
	key, copied, err := b.copyCached(protoPath, template, output)
```

What confused me was that `found == true` would return `err` which I assume is `nil` if the generated files are cached.